Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using UnityEngine;
+
+namespace Player
+{
+    public class PlayerMovement : MonoBehaviour
+    {
+        public float movingSpeed;
+        
+        [Range(0, 1)]
+        public float rotationSmoothness;
+        public float velocitySmoothTime;
+
+        private Animator _animator;
+        private readonly int _isMoving = Animator.StringToHash("IsMoving");
+        
+        private Vector3 _acceleration = Vector3.zero;
+        private Vector3 _velocity = Vector3.zero;
+
+        private void Awake()
+        {
+            _animator = GetComponent<Animator>();
+        }
+
+        private void Update()
+        {
+            var horizontalInput = Input.GetAxisRaw("Horizontal");
+            var verticalInput = Input.GetAxisRaw("Vertical");
+            var direction = new Vector3(horizontalInput, 0, verticalInput);
+            var shouldMove = direction != Vector3.zero;
```

Comment:
floating-point equality checks are unreliable
you should use magnitude checks instead